Mage/src/main/java/mage/abilities/effects/common/combat/CantAttackControlledPlaneswalkerUnlessPayAllEffect.java

```diff
@@ -0,0 +1,61 @@
+
+package mage.abilities.effects.common.combat;
+
+import mage.abilities.Ability;
+import mage.abilities.costs.Cost;
+import mage.abilities.effects.PayCostToAttackBlockEffectImpl;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+/**
+ * @author Susucr
+ */
+public class CantAttackControlledPlaneswalkerUnlessPayAllEffect extends PayCostToAttackBlockEffectImpl {
+
+    private final FilterCreaturePermanent filterCreaturePermanent;
+
+    public CantAttackControlledPlaneswalkerUnlessPayAllEffect(Cost cost) {
+        this(cost, null);
```

Comment:
A bit strange to use null as a special case, rather than the static filter for creatures

Reply:
Uhm indeed. That was copied from `CantAttackYouUnlessPayAllEffect`.
I may move a less generic effect to OnakkeOathkeeper file anyway.

Reply:
Ok I did end up extending CantAttackYouUnlessPayAllEffect for Onakke Oathkeeper sake.

Cleaned the null filter using a default static filter instead.
Also cleaned up the text generation for the simple ones.